extensions/interactions/MultipleChoiceInput/protractor.js

```diff
@@ -19,16 +19,35 @@
 
 var forms = require(process.cwd() + '/core/tests/protractor_utils/forms.js');
 
-// The members of richTextInstructionsArray are functions, one for each option,
-// which will each be passed a 'handler' that they can use to edit the
+// The customizeInstructionsArray contains two elements, The first element is
+// an object, which contains information regrading the type of customization to
+// be done on the interaction(i.e intialize or create interaction, add elements,
+// delete elements). The second element of customizeInstructionsArray is an
+// array, whose elements containing instructions regarding editing of rich text
+// area of options. The members of this array are functions, one for each
+// option, which will each be passed a 'handler' that they can use to edit the
 // rich-text area of the option, for example by
 //   handler.appendUnderlineText('emphasised');
-var customizeInteraction = function(elem, richTextInstructionsArray) {
-  forms.ListEditor(elem).setLength(richTextInstructionsArray.length);
-  for (var i = 0; i < richTextInstructionsArray.length; i++) {
-    var richTextEditor = forms.ListEditor(elem).editItem(i, 'RichText');
-    richTextEditor.clear();
-    richTextInstructionsArray[i](richTextEditor);
+var customizeInteraction = function(elem, customizeInstructionsArray) {
```

Comment:
Instead of making the arguments more bulky, I suggest that you keep the existing argument but perhaps add a few more. If those additional args are unspecified, treat it as a "set". But you could also add two optional args: a command, and an index. If the command is 'delete', verify that richTextInstructionsArray is null, and delete the index. If the command is 'add', verify that richTextInstructionsArray is not null, and contains a single option. If the command is 'edit', verify that richTextInstructionsArray is not null, and contains a single option, and that the index is valid.

In general let's try to avoid defining complex objects with subfields and actions -- it buries too much of the complexity within the object. Also for local changes like this it's probably better to keep as much of the existing code as possible rather than do a full-scale refactor.

Reply:
done

Reply:
done